draft-ietf-quic-recovery.md

```diff
@@ -162,43 +162,209 @@ latency before a userspace QUIC receiver processes a received packet.
+section provides a description of these algorithms. Estimating the network
+round-trip time (RTT) is critical to these algorithms and is described first.
+
+## Computing the RTT estimate
+
+(To be filled)
+
+
+## Ack-based Detection
+
+Ack-based loss detection implements the spirit of TCP's Fast Retransmit
+{{!RFC5681}}, Early Retransmit {{!RFC5827}}, FACK, and SACK loss recovery
+{{!RFC6675}}. This section provides an overview of how these algorithms are
+implemented in QUIC.
+
+### Fast Retransmit
+
+An unacknowledged packet is marked as lost when an acknowledgment is received
+for a packet that was sent a threshold number of packets (kReorderingThreshold)
+after the unacknowledged packet. Receipt of the ack indicates that a later
+packet was received, while kReorderingThreshold provides some tolerance for
+reordering of packets in the network. 
+
+The RECOMMENDED value for kReorderingThreshold is 3.
+
+We derive this default from recommendations for TCP loss recovery {{!RFC5681}}
+{{!RFC6675}}. It is possible for networks to exhibit higher degrees of
+reordering, causing a sender to spuriously detect losses. Spurious loss
+detection leads to unnecessary retransmissions and may result in degraded
+performance due to the actions of the congestion controller upon detecting
+loss. Implementers MAY use algorithms developed for TCP, such as TCP-NCR
+{{!RFC4653}}, to improve QUIC's reordering resilience, though care should be
+taken to map TCP specifics to QUIC correctly. Similarly, using time-based loss
+detection to deal with reordering, such as in PR-TCP, should be more readily
+usable in QUIC. Making QUIC deal with such networks is important open research,
+and implementers are encouraged to explore this space.
+
+### Early Retransmit
+
+Unacknowledged packets close to the tail may have fewer than
+kReorderingThreshold number of packets sent after them. Loss of such packets
+cannot be detected via Fast Retransmit. To enable ack-based loss detection of
+such packets, receipt of an acknowledgment for the last outstanding packet
+triggers the Early Retransmit process, as follows.
+
+If there are unacknowledged packets still pending, they ought to be
+retransmitted, since there are no subsequent packets to trigger a Fast
+Retransmit. To compensate for the reduced reordering resilience, the sender
+SHOULD set an alarm for a small period of time. If the unacknowledged packets
+are not acknowledged during this time, then these packets are marked as lost.
+
+An endpoint SHOULD set the alarm such that a packet is marked as lost no earlier
+than 1.25 * max(SRTT, latest_RTT) since when it was sent. 
+
+Using max(SRTT, latest_RTT) protects from the two following cases:
+
+* the latest RTT sample is lower than the SRTT, perhaps due to reordering where
+packet whose ack triggered the Early Retransit process encountered a shorter
+path;
+
+* the SRTT is lower than the latest RTT sample, perhaps due to a sustained
```

Comment:
nit: I think it's easier to read if it's parallel and the latest RTT comes first in both statements, since you're mostly describing why the latest_rtt would have a smaller/larger value.